go-controller/pkg/ovn/egressip.go

```diff
@@ -1740,8 +1748,10 @@ func (oc *Controller) addNodeForEgress(node *v1.Node) error {
 	v4Addr, v6Addr := getNodeInternalAddrs(node)
 	v4ClusterSubnet, v6ClusterSubnet := getClusterSubnets()
 	if err := oc.createDefaultNoRerouteNodePolicies(v4Addr, v6Addr, v4ClusterSubnet, v6ClusterSubnet); err != nil {
+		oc.addEgressIPNodeFailed.Store(node.Name, node)
 		return err
 	}
+	oc.addEgressIPNodeFailed.Delete(node.Name)
 	if err := oc.initEgressIPAllocator(node); err != nil {
```

Comment:
unrelated question: do we need to worry about us not retrying failures in `initEgressIPAllocator` in line 1755 ? If so, that is likely unrelated to this PR, agree? If anything, maybe we shold add a TODO here, together with a bz to track it. wdyt?

Reply:
this is good question I was wondering about the same too I did look at the code that does the allocation and I couldn't be 100% sure if we should retry allocation errors or not, not sure if retry can recover allocation failures let me look at it again and come back

Reply:
so I checked again the allocator fails for bad configuration hence I don't think retry will make sense with bad configuration, agree ?

Reply:
ack. thanks for following up on that!